packages/core/reference-data-reader/src/index.ts

```diff
@@ -13,37 +13,53 @@ export interface RoundData {
 }
 
 export type ReferenceDataPrice = (
+  network: string,
   contractAddress: string,
   multiply: number,
   meta?: Record<string, unknown>,
 ) => Promise<number>
 
-export type ReferenceDataRound = (contractAddress: string) => Promise<RoundData>
+export type ReferenceDataRound = (network: string, contractAddress: string) => Promise<RoundData>
 
 export const getLatestAnswer: ReferenceDataPrice = async (
+  network,
   contractAddress: string,
   multiply: number,
   meta?: Record<string, unknown>,
 ): Promise<number> => {
-  if (!meta || !meta.latestAnswer) return getRpcLatestAnswer(contractAddress, multiply)
+  if (!meta || !meta.latestAnswer) return getRpcLatestAnswer(network, contractAddress, multiply)
 
   return (meta.latestAnswer as number) / multiply
 }
 export const getRpcLatestAnswer: ReferenceDataPrice = async (
+  network,
   contractAddress: string,
   multiply: number,
 ): Promise<number> => {
-  const rpcUrl = util.getRequiredEnv('RPC_URL')
+  const rpcUrl = getRpcUrl(network)
   const provider = new ethers.providers.JsonRpcProvider(rpcUrl)
   const aggregator = AggregatorInterfaceFactory.connect(contractAddress, provider)
   return (await aggregator.latestAnswer()).div(multiply).toNumber()
 }
 
 export const getRpcLatestRound: ReferenceDataRound = async (
+  network,
   contractAddress: string,
 ): Promise<RoundData> => {
-  const rpcUrl = util.getRequiredEnv('RPC_URL')
+  const rpcUrl = getRpcUrl(network)
   const provider = new ethers.providers.JsonRpcProvider(rpcUrl)
   const aggregator = AggregatorV2V3InterfaceFactory.connect(contractAddress, provider)
   return await aggregator.latestRoundData()
 }
+
+export const getRpcUrl = (network: string): string => {
+  // Backwards compatability for RPC_URL
+  const rpcURL = util.getEnv('RPC_URL')
+  if (rpcURL) {
```

Comment:
Can we check this as a fallback if `${network.toUpperCase}_RPC_URL` isn't set?

I see node ops just putting RPC_URL by habit, which would break the other network_RPC_URL support.

Though I guess we don't get to really utilize the `getRequiredEnv` helper?

Reply:
Yeah, won't get to use `getRequiredEnv`, but that's not a huge deal. I'll tweak for that.